src/main/java/de/tum/in/www1/artemis/repository/CompetencyRelationRepository.java

```diff
@@ -33,4 +33,10 @@ public interface CompetencyRelationRepository extends JpaRepository<CompetencyRe
             """)
     Set<CompetencyRelation> findAllByCourseId(@Param("courseId") Long courseId);
 
+    @Query("""
+                    SELECT count(cr)
+                    FROM CompetencyRelation cr
+                    WHERE cr.headCompetency.course.id = :courseId OR cr.tailCompetency.course.id = :courseId
```

Comment:
Is it really possible that the head of a competency relation is in another course than the tail? When does this happen?

Reply:
I know we discussed this already in private, but here for everyone: For now LPs only exist within a course. We're planning to extend this feature to Artemis-wide LPs. This query takes this already into account.